packages/astro/src/i18n/vite-plugin-i18n.ts

```diff
@@ -0,0 +1,26 @@
+import * as vite from 'vite';
+import type { AstroSettings } from '../@types/astro.js';
+import type { Logger } from '../core/logger/core.js';
+
+const virtualModuleId = 'astro:i18n';
+const resolvedVirtualModuleId = '\0' + virtualModuleId;
+
+type AstroInternalization = {
+	settings: AstroSettings;
+	logger: Logger;
+};
+
+export default function astroInternalization({ settings }: AstroInternalization): vite.Plugin {
+	return {
+		name: 'astro:i18n',
+		load(id) {
+			if (id === resolvedVirtualModuleId) {
+				return `
+					import { getI18nBaseUrl as getI18nBaseUrlInternal } from "astro/i18n";
+								
+					export getI18nUrl = (locale) => getI18nBaseUrlInternal(locale, ${settings.config});
```

Comment:
I'm not sure if this will work as `settings.config` isn't necessarily completely serializable as a string, and the logger isn't passed in too 👀

Reply:
> I'm not sure if this will work as settings.config isn't necessarily completely serializable as a string

Where am I expecting to retrieve this value then? 

> and the logger isn't passed in too 👀

Cheers, I'll fix it when I will tackle this part

Reply:
Since you're only reading the `base` and the truthy value of `config.experimental.i18n`, you can read both of them outside of this string, and then inject them individually into this string, since both are serializable (with `JSON.stringify(someValue)`).

But I'm not sure a logger is serializable though. You might need to resort to `console.warn` instead.

Reply:
Ah ok, I thought you were talking about something else. Yes that will be done when I tackle the virtual module